lms/djangoapps/certificates/api.py

```diff
@@ -776,3 +780,92 @@ def certificate_info_for_user(user, course_id, grade, user_is_allowlisted, user_
 def certificate_status_for_student(student, course_id):
     """This returns a dictionary with a key for status, and other information."""
     return _certificate_status_for_student(student, course_id)
+
+
+def auto_certificate_generation_enabled():
+    return _AUTO_CERTIFICATE_GENERATION.is_enabled()
+
+
+def _enabled_and_instructor_paced(course):
+    if auto_certificate_generation_enabled():
+        return not course.self_paced
+    return False
+
+
+def can_show_certificate_available_date_field(course):
+    return _enabled_and_instructor_paced(course)
+
+
+def can_show_certificate_message(course, student, course_grade, certificates_enabled_for_course):
+    """
+    Returns True if a course certificate message can be shown
+    """
+    auto_cert_gen_enabled = auto_certificate_generation_enabled()
+    has_active_enrollment = CourseEnrollment.is_enrolled(student, course.id)
+    certificates_are_viewable = certificates_viewable_for_course(course)
+    is_beta_tester = access.is_beta_tester(student, course.id)
+    has_passed_or_is_allowlisted = _has_passed_or_is_allowlisted(course, student, course_grade)
+
+    return (
+        (auto_cert_gen_enabled or certificates_enabled_for_course) and
+        has_active_enrollment and
+        certificates_are_viewable and
+        has_passed_or_is_allowlisted and
+        (not is_beta_tester)
+    )
+
+
+def _course_uses_available_date(course):
+    """Returns if the course has an certificate_available_date set and that it should be used"""
+    if settings.FEATURES.get("ENABLE_V2_CERT_DISPLAY_SETTINGS"):
```

Comment:
It'd be delightful if this was a constant, but I support "move the code now and improve on it later". :-)